server/src/main/java/org/opensearch/index/mapper/StarTreeMapper.java

```diff
@@ -262,14 +262,19 @@ private Metric getMetric(String name, Map<String, Object> metric, Mapper.TypePar
                 .collect(Collectors.toList());
             metric.remove(STATS);
             if (metricStrings.isEmpty()) {
-                metricTypes = new ArrayList<>(StarTreeIndexSettings.DEFAULT_METRICS_LIST.get(context.getSettings()));
-            } else {
-                Set<MetricStat> metricSet = new LinkedHashSet<>();
-                for (String metricString : metricStrings) {
-                    metricSet.add(MetricStat.fromTypeName(metricString));
+                metricStrings = new ArrayList<>(StarTreeIndexSettings.DEFAULT_METRICS_LIST.get(context.getSettings()));
+            }
+            // Add all required field initially
+            Set<MetricStat> metricSet = new LinkedHashSet<>(MetricStat.getRequiredMetrics());
+            for (String metricString : metricStrings) {
```

Comment:
nit: metricType?

Reply:
Already metricTypes is present in context and hence can't change the list name. metricString fits better than metricType to loop through metricStrings